BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ReflectionCredentialEnvelope.cs

```diff
@@ -19,6 +19,12 @@
     /// </remarks>
     internal class ReflectionCredentialEnvelope : CredentialEnvelope
     {
+#if REDFIELD
+        private static readonly string AzureCoreAssemblyName = "Azure.Identity.ILRepack";
```

Comment:
Can this be a private const?

Reply:
Kept it intentionally as readonly to use reflection if there be a need in this area. We had to prepare this change as there was no option for reflection. Please correct me if this approach is not the right thing to do.

Reply:
Got it, consider removing the "readonly" part to emphasize the semantic (and also prevent the compiler from optimizing too much).

Reply:
https://fluentprogrammer.com/volatile-keyword-in-c-and-c-plus-plus/